### PR TITLE
added support for ng private endpoints

### PR DESCRIPTION
**Pull request by arahamad**

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -154,25 +154,29 @@ type VPCProviderConfig struct {
...
+	G2ResourceGroupID    string `toml:"g2_resource_group_id"`
+	G2VPCAPIGeneration   int    `toml:"g2_vpc_api_generation" envconfig:"G2_VPC_API_GENERATION"`
+	G2APIVersion         string `toml:"g2_api_version,omitempty" envconfig:"G2_VPC_API_VERSION"`
 
 	Encryption      bool   `toml:"encryption"`
 	VPCTimeout      string `toml:"vpc_api_timeout,omitempty" envconfig:"VPC_API_TIMEOUT"`
 	MaxRetryAttempt int    `toml:"max_retry_attempt,omitempty" envconfig:"VPC_RETRY_ATTEMPT"`
 	MaxRetryGap     int    `toml:"max_retry_gap,omitempty" envconfig:"VPC_RETRY_INTERVAL"`
+	// IKSTokenExchangePrivateURL, for private cluster support hence using for all cluster types
+	IKSTokenExchangePrivateURL string `toml:"iks_token_exchange_endpoint_private_url"`
```
Comment (prgavali) [inline config/config.go]: Can we make this as `IKSTokenExchangePrivateURL string `toml:"iks_token_exchange_endpoint_private_url" toml:"gc_token_exchange_endpoint_private_url" toml:"g2_token_exchange_endpoint_private_url"`

For third party vendor , they can put the private endpoint in one of these three properties

File: volume-providers/vpc/provider/provider.go [hunk 2]
```diff
@@ -78,14 +78,13 @@ func NewProvider(conf *config.Config, logger *zap.Logger) (local.Provider, error
 	}
 
 	//Do config validation and enable only one generationType (i.e VPC-Classic | VPC-NG)
-	gcConfigFound := (conf.VPC.EndpointURL != "") && (conf.VPC.TokenExchangeURL != "") && (conf.VPC.APIKey != "") && (conf.VPC.ResourceGroupID != "")
-	g2ConfigFound := (conf.VPC.G2EndpointURL != "") && (conf.VPC.G2TokenExchangeURL != "") && (conf.VPC.G2APIKey != "") && (conf.VPC.G2ResourceGroupID != "")
+	gcConfigFound := (conf.VPC.EndpointURL != "" || conf.VPC.PrivateEndpointURL != "") && (conf.VPC.TokenExchangeURL != "" || conf.VPC.IKSTokenExchangePrivateURL != "") && (conf.VPC.APIKey != "") && (conf.VPC.ResourceGroupID != "")
+	g2ConfigFound := (conf.VPC.G2EndpointPrivateURL != "") && (conf.VPC.IKSTokenExchangePrivateURL != "") && (conf.VPC.G2APIKey != "") && (conf.VPC.G2ResourceGroupID != "")
```
Comment (prgavali) [inline volume-providers/vpc/provider/provider.go]: Do we need to check `conf.VPC.G2EndpointURL != "" || conf.VPC.G2EndpointPrivateURL != ""` ?
You are checking this in gcConfigFound

File: config/config.go (hunk 1 above)
Comment (arahamad) [reply config/config.go]: I did not see this is working

File: volume-providers/vpc/provider/provider.go (hunk 2 above)
Comment (arahamad) [reply volume-providers/vpc/provider/provider.go]: added

**Comment by arahamad** [comment]
tested flow for ng and gc
including private . endpoint and mixing config

**Pull request merged**